bundles/org.openhab.voice.actiontemplatehli/src/main/java/org/openhab/voice/actiontemplatehli/internal/ActionTemplateInterpreter.java

```diff
@@ -0,0 +1,1159 @@
+                            : new String[] {};
+                    var unprocessedTokensTemplate = Arrays.copyOfRange(tokensTemplate, processedIndex,
+                            tokensTemplate.length);
+                    Collections.reverse(Arrays.asList(unprocessedTokens));
+                    Collections.reverse(Arrays.asList(unprocessedTags));
+                    Collections.reverse(Arrays.asList(unprocessedTokensTemplate));
+                    if (DYNAMIC_PLACEHOLDER_SYMBOL.equals(unprocessedTokens[0])) {
+                        // here dynamic placeholder should be at the end, but if it's also at the beginning we should
+                        // abort
+                        logger.warn("Using multiple dynamic placeholders is not supported");
+                        return NLPTokenComparisonResult.ZERO;
+                    }
+                    var partialScoreResult = compareTokens(unprocessedTokens, unprocessedTags,
+                            unprocessedTokensTemplate);
+                    if (NLPTokenComparisonResult.ZERO.equals(partialScoreResult)) {
+                        return NLPTokenComparisonResult.ZERO;
+                    } else {
+                        var dynamicSpan = partialScoreResult.dynamicSpan;
+                        if (dynamicSpan == null) {
+                            logger.error(
+                                    "dynamic span missed, this should never happen, please open an issue; aborting");
+                            return NLPTokenComparisonResult.ZERO;
+                        }
+                        return new NLPTokenComparisonResult(score + partialScoreResult.score,
+                                new Span(i, tokens.length - (dynamicSpan.getStart())));
+                    }
+                }
+                if (tokenAlternative.equals(token)) {
+                    isMatch = true;
+                    break;
+                }
+            }
+            if (isMatch) {
+                processedIndex++;
+                score++;
+            } else if (tag != null && optionalLanguageTags.contains(tag)) {
+                logger.debug("part '{}' tagged as '{}' skipped", token, tag);
+            } else {
+                return NLPTokenComparisonResult.ZERO;
+            }
+        }
+        return new NLPTokenComparisonResult(score, null);
+    }
+
+    private String[] splitString(String template, String regex) {
+        return Arrays.stream(template.split(regex)).map(String::trim).toArray(String[]::new);
+    }
+
+    private void invalidate() {
+        logger.debug("Invalidate cached item data");
+        nlpItemMaps = null;
+    }
+
+    private void reloadConfigs() {
+        optionalLanguageTags = Arrays.stream(this.config.optionalLanguageTags.split(",")).filter(i -> !i.isEmpty())
+                .collect(Collectors.toList());
+        cacheMap.invalidateAll();
+        try {
+            tokenizer = getTokenizer();
+        } catch (IOException e) {
```

Comment:
I think it will be better to catch the exception directly in method getTokenizer.